app/experimenter/experiments/tests/test_models.py

```diff
@@ -1527,6 +1534,57 @@ def test_is_json_string_type_returns_true_for_json_string_type(self):
         self.assertTrue(self.pref.is_json_string_type)
 
 
+class TestRolloutPreference(TestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.experiment = ExperimentFactory.create()
+        self.pref = RolloutPreference()
+        self.pref.experiment = self.experiment
+        self.pref.pref_type = Experiment.PREF_TYPE_BOOL
+        self.pref.pref_value = "true"
+        self.pref.pref_name = "pref_name"
+        self.pref.save()
+
+    def test_unique_pref_name_constraint_is_violated(self):
+
+        pv2 = RolloutPreference()
+        pv2.pref_type = Experiment.PREF_TYPE_BOOL
+        pv2.pref_value = "false"
+        pv2.pref_name = "pref_name"
+        pv2.experiment = self.experiment
+
+        with self.assertRaises(IntegrityError):
+            pv2.save()
+
+    def test_valid_preference_are_associated_to_experiment(self):
+
+        pv2 = RolloutPreference()
+        pv2.pref_type = Experiment.PREF_TYPE_BOOL
+        pv2.pref_value = "false"
+        pv2.pref_name = "pref2_name"
+        pv2.experiment = self.experiment
+
+        pv2.save()
+
+        self.assertTrue(self.experiment.preferences.count, 2)
+
+    def test_valid_pref_is_able_to_update_itself(self):
+        self.pref.value = 6
+
+        self.pref.save()
+
+        self.assertTrue(self.experiment.preferences.count, 1)
+
+    def test_is_json_string_type_returns_false(self):
+        self.assertFalse(self.pref.is_json_string_type)
```

Comment:
The test should setup the data it needs to pass inside the test otherwise we could change some defaults and this would suddenly fail.